dashboard/app/models/pd/application/teacher_application.rb

```diff
@@ -84,12 +84,12 @@ class TeacherApplication < ApplicationBase
 
     has_many :emails, class_name: 'Pd::Application::Email', foreign_key: 'pd_application_id'
 
+    before_validation :set_course_from_program, unless: -> {program.nil?}
```

Comment:
I just moved this line from below for my brain to comprehend the order in which these things happen